service/templates/cloudformation/security_groups.yaml

```diff
@@ -0,0 +1,49 @@
+{{define "security_groups" }}
+  MasterSecurityGroup:
+    Type: AWS::EC2::SecurityGroup
+    Properties:
+      GroupDescription:  {{ .MasterSecurityGroupName }}
+      VpcId: {{ .VPCID }}
+      Tags:
+        - Key: Name
+          Value:  {{ .MasterSecurityGroupName }}
+
+    WorkerSecurityGroup:
+      Type: AWS::EC2::SecurityGroup
+      Properties:
+        GroupDescription: {{ .WorkerSecurityGroupName }}
+        VpcId: {{ .VPCID }}
+        SecurityGroupIngress:
+      {{ range .WorkerSecurityGroupRules }}
+      - 
+        IpProtocol: {{ .Protocol }}
+        FromPort: {{ .Port }}
+        ToPort: {{ .Port }}
+        {{ if .SourceCIDR }}
+        CidrIp: {{ .SourceCIDR }}
+        {{ else }}
+        SourceSecurityGroupName: {{ .SourceSecurityGroupName }}
+        {{ end }}
+      {{ end }}
```

Comment:
The conditional logic here matches what we do now. So if there is a SourceCIDR it takes precedence. But its a bit untidy. I did think of having separate slices for CIDR and non-CIDR based rules. WDYT?

Reply:
IMO it's better this way, it's not too much logic and having separate slices would make things less clear both in the template and the adapter.

Reply:
Thanks, that makes sense and I'll use the same approach for the master rules.